js/client-battle.js

```diff
@@ -1117,20 +1117,43 @@
 		toggleIgnoreSpects: function (e) {
 			this.battle.ignoreSpects = !!e.currentTarget.checked;
 			this.battle.add('Spectators ' + (this.battle.ignoreSpects ? '' : 'no longer ') + 'ignored.');
+			var $messages = $('.battle-log').find('[class*="chatmessage-"]').not('small:contains(\u2605)');
+			if (!$messages.length) return;
+			if (this.battle.ignoreSpects) {
+				$messages.hide();
+			} else {
+				$messages.show();
+			}
 		},
 		toggleIgnoreNicks: function (e) {
 			this.battle.ignoreNicks = !!e.currentTarget.checked;
 			Tools.prefs('ignorenicks', this.battle.ignoreNicks);
 			this.battle.add('Nicknames ' + (this.battle.ignoreNicks ? '' : 'no longer ') + 'ignored.');
 			this.toggleNicknames(this.battle.mySide);
 			this.toggleNicknames(this.battle.yourSide);
+
+			var $log = $('.battle-log .inner');
+			$log = $log.find('span.pokemon');
+			if (this.battle.ignoreNicks) {
+				$log.addClass('hidenick');
```

Comment:
You should probably add the class to the log as a whole, instead of individual nicks.